honeybee_energy/construction.py

```diff
@@ -242,7 +274,7 @@ def materials(self, mats):
             if not isinstance(mats, tuple):
                 mats = tuple(mats)
         except TypeError:
-            raise TypeError('Expected list for construction materials. '
+            raise TypeError('Expected list or tuple for construction materials. '
                             'Got {}'.format(type(mats)))
         for mat in mats:
             assert isinstance(mat, _EnergyMaterialOpaqueBase), 'Expected opaque energy' \
```

Comment:
I'm unclear what you are suggesting here.  You just think the generic warning message `mats is not iterable` is sufficient for the case that that someone does not input an ilterable here? So it's not necessary to say 'Expected list or tuple for construction materials.  Got {}'

Reply:
What I mean is if you check for every material to be an instance of valid material it will already cover the case for input not being iterable. Basically it will throw an error in line 279. I agree that it won't be as clean as your error message so maybe what you have there is just as good.